pkg/pillar/hypervisor/xen.go

```diff
@@ -341,110 +338,82 @@ func (ctx xenContext) CreateDomConfig(domainName string, config types.DomainConf
 
 func (ctx xenContext) Create(domainName string, xenCfgFilename string, config *types.DomainConfig) (int, error) {
 	log.Infof("xlCreate %s %s\n", domainName, xenCfgFilename)
-	cmd := "xl"
-	args := []string{
-		"create",
-		xenCfgFilename,
-		"-p",
-	}
-	stdoutStderr, err := wrap.Command(cmd, args...).CombinedOutput()
+	_, err := containerd.LKTaskLaunch(domainName, "xen-tools", config,
+		[]string{"xl", "create", xenCfgFilename, "-p", "-F", "-c"})
 	if err != nil {
 		log.Errorln("xl create failed ", err)
-		log.Errorln("xl create output ", string(stdoutStderr))
-		return 0, fmt.Errorf("xl create failed: %s\n",
-			string(stdoutStderr))
+		return 0, fmt.Errorf("xl create failed")
 	}
 	log.Infof("xl create done\n")
 
-	args = []string{
-		"domid",
-		domainName,
-	}
-	stdoutStderr, err = wrap.Command(cmd, args...).CombinedOutput()
+	stdOut, stdErr, err := containerd.CtrExec(domainName,
+		[]string{"xl", "domid", domainName})
 	if err != nil {
 		log.Errorln("xl domid failed ", err)
-		log.Errorln("xl domid output ", string(stdoutStderr))
-		return 0, fmt.Errorf("xl domid failed: %s\n",
-			string(stdoutStderr))
+		log.Errorln("xl domid output ", string(stdOut), string(stdErr))
+		return 0, fmt.Errorf("xl domid failed: %s %s", string(stdOut), string(stdErr))
 	}
-	res := strings.TrimSpace(string(stdoutStderr))
+	res := strings.TrimSpace(string(stdOut))
 	domainID, err := strconv.Atoi(res)
 	if err != nil {
-		log.Errorf("Can't extract domainID from %s: %s\n", res, err)
-		return 0, fmt.Errorf("Can't extract domainID from %s: %s\n", res, err)
+		log.Errorf("Can't extract domainID from %s: %s", res, err)
+		return 0, fmt.Errorf("Can't extract domainID from %s: %s", res, err)
 	}
 	return domainID, nil
 }
 
 func (ctx xenContext) Start(domainName string, domainID int) error {
 	// Disable offloads for all vifs
-	stdoutStderr, err := wrap.Command("/etc/xen/scripts/disable-vif-features",
-		strconv.Itoa(domainID)).CombinedOutput()
+	stdOut, stdErr, err := containerd.CtrExec(domainName,
+		[]string{"/etc/xen/scripts/disable-vif-features", strconv.Itoa(domainID)})
```

Comment:
What are we doing here? This appears to be much more fundamental than refactoring; it appears to change how we do xen domain start and stop. Is it? If so, it really should live in its own PR and be written up as a design doc, so that someone coming into it and wanting to help can understand the design before hitting the code.

Reply:
There's actually a design doc that got published and reviewed @deitch  https://wiki.lfedge.org/display/EVE/Tasks+subsystem 

As to whether this can be in its own PR -- I don't think I can test the other refactoring with this final piece. Hence making it part of this PR (but note that it is in a separate commit).